src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -52,8 +52,9 @@ void BulkReader::run() {
         if (result >= 0) {
             Trace process("BulkReader process packet");
             //qDebug() << "Read" << result << "bytes, pointer:" << data;
-            QByteArray outData((char*)data, transferred);
-            emit incomingData(outData, mixxx::Time::elapsed());
+            QByteArray byteArray = QByteArray::fromRawData(
+                    reinterpret_cast<char*>(data), transferred);
+            emit incomingData(byteArray, mixxx::Time::elapsed());
```

Comment:
Wait, is this safe? BulkReader is running in its own thread, not the controller polling thread.

Reply:
Also, why is it running in its own thread...

Reply:
Should we have all controllers polling in the same thread? This isn't an option with libhss1394 though:
```
	class HSS1394_CLASS_CONVENTION ChannelListener {
		public:
			// :: Process is called when data has arrived. The arguments present
			//! the data. On completion of this call, pBuffer is no longer valid.
			//! This call will occur inside a separate thread. The implementer must
			//! ensure concurrency control issues are dealt with.
			virtual void Process(const uint8 *pBuffer, uint uBufferSize) { }
```

Reply:
ping

Reply:
But is it safe to not make a deep copy passing data between threads?

Reply:
It is save to pass implicit shared objects between threads. However we have here the issue that we trick that feature by using fromRawData() this will alter the message if a new telegram arrived before to old one is processed. So we must not use fromRawData() here.

Reply:
Using QByteArray::fromRawData() would be safe here. The pointer will never leave the current thread's context and the data is cloned when making a copy of the QByteArray.

But since we always pass it to another thread cloning the data immediately does not have any performance impact. In any case the data is cloned exactly once.